sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonSQLMetrics.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.python
+
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.metric.{SQLMetric, SQLMetrics}
+
+private[python] trait PythonSQLMetrics { self: SparkPlan =>
+
+  override lazy val metrics: Map[String, SQLMetric] = PythonSQLMetricsNames.names.map {
+    case (name: String, description: String) =>
+      (name, SQLMetrics.createSizeMetric(sparkContext, description))
+  }
+
+  lazy val pythonMetrics = PythonSQLMetricsNames.names.map {
```

Comment:
isn't it just `metrics.toSeq`?

Reply:
Thanks for reviewing. Good point, I have refactored this part.